extensions/amp-form/amp-form.md

```diff
@@ -291,6 +296,38 @@ This shows all validation errors on all invalid inputs when the form is submitte
 #### As You Go
 This allows your user to see validation messages as they're interacting with the input, if the email they typed is invalid they'll see the error right away and once fixed the error goes away.
 
+## Variable Substitutions
+__(<a href="https://www.ampproject.org/docs/reference/experimental.html">experimental</a>)__
+
+`amp-form` allows [variable substitutions](../../spec/amp-var-substitutions.md) for inside hidden inputs with `default-value` attribute. On each submission, `amp-form` finds all `input[type=hidden]` inside the form and apply variable substitutions to `default-value` attribute and fill the result in the input `value`.
```

Comment:
Yeah sure, would this look like the following:
```html
<input value="can: CANONICAL_URL, rand: RANDOM" data-amp-replace="CANONICAL_URL RANDOM">
```

`amp-form` would then call a modified `maybeExpandLink` with the expected attribute to replace (`value` in this case and `href` in links case).

@rudygalfi @cramforce does this sound better?